File: app/src/main/java/vn/loitp/app/activity/customviews/menu/residemenu/HomeFragment.java
package vn.loitp.app.activity.customviews.menu.residemenu;

import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.annotation.LogTag;
import com.core.base.BaseFragment;
import com.views.menu.residemenu.ResideMenu;

import vn.loitp.app.R;

@LogTag("HomeFragment")
public class HomeFragment extends BaseFragment {
    private ResideMenu resideMenu;

    @Override
    protected int setLayoutResourceId() {
        return R.layout.reside_menu_home;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setUpViews();
    }

    private void setUpViews() {
        final ResideMenuActivity parentActivity = (ResideMenuActivity) getActivity();
        resideMenu = parentActivity.getResideMenu();

        getFrmRootView().findViewById(R.id.btOpenMenu).setOnClickListener(view -> resideMenu.openMenu(ResideMenu.DIRECTION_LEFT));

        FrameLayout ignored_view = getFrmRootView().findViewById(R.id.ignored_view);
        resideMenu.addIgnoredView(ignored_view);
    }

}

File: app/src/main/java/vn/loitp/app/activity/customviews/menu/residemenu/HomeFragment.kt
package vn.loitp.app.activity.customviews.menu.residemenu

import android.os.Bundle
import android.view.View
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFragment
import com.loitpcore.views.menu.residemenu.ResideMenu
import kotlinx.android.synthetic.main.reside_menu_home.*
import vn.loitp.app.R

@LogTag("HomeFragment")
class HomeFragment : BaseFragment() {

    private var resideMenu: ResideMenu? = null

    override fun setLayoutResourceId(): Int {
        return R.layout.reside_menu_home
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setUpViews()
    }

    private fun setUpViews() {
        val parentActivity = activity as ResideMenuActivity
        resideMenu = parentActivity.resideMenu

        btOpenMenu.setOnClickListener { view: View? ->
            resideMenu?.openMenu(ResideMenu.DIRECTION_LEFT)
        }

        resideMenu?.addIgnoredView(layoutIgnoredView)
    }
}
